services/navigation/builtin/builtin.go

```diff
@@ -575,7 +581,49 @@ func (svc *builtIn) waypointIsDeleted() bool {
 func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{}) ([]*spatialmath.GeoObstacle, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return svc.obstacles, nil
+
+	gobs := svc.obstacles
+
+	for _, detector := range svc.motionCfg.ObstacleDetectors {
+		visSvc, ok := svc.visionServices[detector.VisionServiceName]
+		if !ok {
+			return nil, fmt.Errorf("vision service with name: %v not found", detector.VisionServiceName)
+		}
+		gp, _, err := svc.movementSensor.Position(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		detections, err := visSvc.GetObjectPointClouds(ctx, detector.CameraName.Name, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		var geoms []spatialmath.Geometry
+		for i, detection := range detections {
+			// a bump or ultrasonic sensor will always return a pointcloud of size 1 along with a
+			// geometry which has no volume
+			var geom spatialmath.Geometry
+			if detection.PointCloud.Size() == 1 {
```

Comment:
This seems like a fragile way to check for vision services of this type.  Is there another way we can do it without checking the size directly?

Reply:
Given a vision name `"obs"` calling `visSvc.Name()` would give us `rdk:service:vision/obs`.
I think instead we can do:
```
			if detection.Geometry.AlmostEqual(
				spatialmath.NewPoint(r3.Vector{position.Point().X, position.Point().Y, position.Point().Z}, ""),
			) {
```